src/media/playback/playback_device.cpp

```diff
@@ -380,12 +380,14 @@ void playback_device::resume()
             return;
 
         auto total_duration = m_reader->query_duration();
-        if (m_last_published_timestamp >= total_duration)
-            m_last_published_timestamp = device_serializer::nanoseconds(0);
-        m_reader->reset();
-        m_reader->seek_to_time(m_last_published_timestamp);
-        while (m_last_published_timestamp != device_serializer::nanoseconds(0) && !m_reader->read_next_data()->is<serialized_frame>());
-
+        {
```

Comment:
Did you verify R&P with the viewer after this change?
If not please do pause on/off.

Reply:
Verified. Including pause-play, stop and play, frame by frame etc...